File: doc/codes/python/examples/api/compute/pattern/focus.py
import numpy as np
from autd3.geometry import Autd3, Geometry
from autd3.units import m, s
from autd3_pattern import focus, wavelength as calc_wavelength

geometry = Geometry([Autd3([0.0, 0.0, 0.0], [1.0, 0.0, 0.0, 0.0])])

target = geometry.center() + np.array([0.0, 0.0, 150.0])
wavelength = calc_wavelength(340 * m / s)
phases = geometry.phase_buffer()

# ANCHOR: api
focus(geometry, target, wavelength, phases)
# ANCHOR_END: api

File: doc/codes/rust/examples/api/compute/pattern/focus.rs
use autd3_rs::geometry::{Autd3, Geometry, offset};
use autd3_rs::units::{m, mm, s};
use autd3_rs_pattern::{focus, wavelength};

fn main() {
    let geometry = Geometry::new(vec![Autd3::default()]);

    let target = geometry.center() + offset(0.0 * mm, 0.0 * mm, 150.0 * mm);
    let wavelength = wavelength(340.0 * m / s);
    let mut phases = geometry.phase_buffer();

    // ANCHOR: api
    focus(&geometry, target, wavelength, &mut phases);
    // ANCHOR_END: api
}
